lib/src/system_cache.dart

```diff
@@ -36,19 +34,19 @@ class SystemCache {
 
   static String defaultDir = (() {
     if (Platform.environment.containsKey('PUB_CACHE')) {
-      return Platform.environment['PUB_CACHE'];
+      return Platform.environment['PUB_CACHE']!;
     } else if (Platform.isWindows) {
       // %LOCALAPPDATA% is preferred as the cache location over %APPDATA%, because the latter is synchronised between
       // devices when the user roams between them, whereas the former is not.
       // The default cache dir used to be in %APPDATA%, so to avoid breaking old installs,
       // we use the old dir in %APPDATA% if it exists. Else, we use the new default location
       // in %LOCALAPPDATA%.
-      var appData = Platform.environment['APPDATA'];
+      var appData = Platform.environment['APPDATA']!;
```

Comment:
We should probably handle the absense of appdata.
```suggestion
      TODO(sigurdm): handle missing APPDATA.
      var appData = Platform.environment['APPDATA']!;
```

Reply:
Done